zerver/models/users.py

```diff
@@ -800,8 +805,8 @@ def has_permission(self, policy_name: str) -> bool:
         assert policy_value == Realm.POLICY_FULL_MEMBERS_ONLY
         return not self.is_provisional_member
 
-    def can_create_public_streams(self) -> bool:
-        return self.has_permission("can_create_public_channel_group")
+    def can_create_public_streams(self, realm: Optional["Realm"] = None) -> bool:
```

Comment:
We can also make it mandatory for all the callers to pass realm, but then we only actually fetch the settings in `register` response.

Reply:
This is probably OK.